packages/components/pie-radio-group/src/index.ts

```diff
@@ -147,17 +160,262 @@ export class PieRadioGroup extends FormControlMixin(RtlMixin(LitElement)) implem
         }
     }
 
+    protected firstUpdated (): void {
+        // Make all radios impossible to tab to
+        this._slottedChildren.forEach((radio) => radio.setAttribute('tabindex', '-1'));
+    }
+
     connectedCallback (): void {
         super.connectedCallback();
         this._abortController = new AbortController();
         const { signal } = this._abortController;
 
         this.shadowRoot?.addEventListener('change', this._handleRadioChange.bind(this), { signal });
+
+        this.addEventListener('focusin', this._handleFocusIn, { signal });
+        this.addEventListener('focusout', this._handleFocusOut, { signal });
+
+        this.addEventListener('keydown', this._handleKeyDown, { signal });
+        document.addEventListener('keydown', this._updateShiftTabState.bind(this), { signal });
+    }
+
+    /**
+     * Updates the state of `_wasShiftTabPressed` based on the last `Tab` key press.
+     *
+     * When a `Tab` key press is detected, this method updates the static property
+     * `wasShiftTabPressed` to indicate whether the `Shift` key was also held during
+     * the event. This information is used to manage focus direction when the radio
+     * group gains focus.
+     *
+     * @param {KeyboardEvent} event - The keyboard event containing information about the key press.
+     *
+     * Example Usage:
+     * - If `Shift+Tab` is pressed, `_wasShiftTabPressed` is set to `true` to signal backward navigation.
+     * - If `Tab` alone is pressed, `_wasShiftTabPressed` is set to `false` for forward navigation.
+     */
+    private _updateShiftTabState (event: KeyboardEvent): void {
+        if (event.key === 'Tab') {
+            PieRadioGroup._wasShiftTabPressed = event.shiftKey;
+        }
+    }
+
+    /**
+     * Handles the `focusin` event to manage focus within the radio group.
+     *
+     * This method determines the appropriate element to focus when the radio group
+     * gains focus. It considers the last navigation action (whether `Shift+Tab` was used)
+     * and focuses the checked option, the first option, or the last option as needed.
+     *
+     * **Behaviour:**
+     * - If the event target is the radio group itself:
+     *   - If an option is already checked, it focuses the checked option.
+     *   - If `Shift+Tab` was pressed, it focuses the last option.
```

Comment:
Where did this requirement come from, out of interest?

Reply:
Actually I see it's mimicking native behaviour

Reply:
However focusing the checked option doesn't seem to be native behaviour, unless you saw differently?

Reply:
It's part of the recommended pattern from W3C:

https://www.w3.org/WAI/ARIA/apg/patterns/radio/

<img width="1025" alt="Screenshot 2024-12-12 at 11 23 40" src="https://github.com/user-attachments/assets/33932c0f-e06d-49fa-bdc4-377ca7d86437" />

Reply:
I have been comparing against this [native example in codepen](https://codepen.io/JamieMaguire/pen/VYZZoYz)